src/model.ts

```diff
@@ -0,0 +1,455 @@
+        document.body.style.setProperty(
+          `${'--jp-code-font-size'}`,
+          String(value) + 'px'
+        );
+      },
+      'border-width': (value: number) => {
+        document.body.style.setProperty(
+          `${'--jp-border-width'}`,
+          String(value) + 'px'
+        );
+      },
+      'border-radius': (value: number) => {
+        document.body.style.setProperty(
+          `${'--jp-border-radius'}`,
+          String(value) + 'px'
+        );
+      },
+      'accent-color': (value: string) => {
+        defineAndApplyPaletteFrom1hexColor(value, 4, '--jp-accent-color');
+      },
+      'border-color': (value: string) => {
+        defineAndApplyPaletteFrom1hexColor(value, 4, '--jp-border-color');
+      },
+      'brand-color': (value: string) => {
+        defineAndApplyPaletteFrom1hexColor(value, 4, '--jp-brand-color');
+      },
+      'error-color': (value: string) => {
+        defineAndApplyPaletteFrom1hexColor(value, 4, '--jp-error-color');
+      },
+      'info-color': (value: string) => {
+        defineAndApplyPaletteFrom1hexColor(value, 4, '--jp-info-color');
+      },
+      'layout-color': (value: string) => {
+        const colorValueRGBAstr = hexToRGBA(value);
+        const colorRGBA64 = rgbaStringToColorRGBA64(colorValueRGBAstr);
+        const colorRGBA64shifted = shiftLuminanceRGBAColor(colorRGBA64);
+        const palette = definePalette(colorRGBA64shifted, 4);
+        applyPalette(palette, '--jp-layout-color');
+
+        const shiftedLuminance = getLuminanceRGBAColor(colorRGBA64shifted);
+        const complementaryL = 1 - shiftedLuminance;
+        const inverseColorHSL = new ColorHSL(
+          getHueRGBAColor(colorRGBA64),
+          getSaturationRGBAColor(colorRGBA64),
+          complementaryL
+        );
+        const inverseColorRGBA64 = hslToRGB(inverseColorHSL, 1);
+        const inversePalette = definePalette(inverseColorRGBA64, 5);
+        applyPalette(inversePalette, '--jp-inverse-layout-color');
+        applyPalette(inversePalette, '--jp-ui-font-color');
+      },
+      'success-color': (value: string) => {
+        const colorValueRGBAstr = hexToRGBA(value);
+        const colorRGBA64 = rgbaStringToColorRGBA64(colorValueRGBAstr);
+        const palette = definePalette(colorRGBA64, 4);
+        applyPalette(palette, '--jp-success-color');
+      },
+      'warn-color': (value: string) => {
+        const colorValueRGBAstr = hexToRGBA(value);
+        const colorRGBA64 = rgbaStringToColorRGBA64(colorValueRGBAstr);
+        const palette = definePalette(colorRGBA64, 4);
+        applyPalette(palette, '--jp-warn-color');
+      }
+    };
+  }
+  public get schema(): any {
+    return this._schema;
+  }
+
+  public get formData(): any {
+    return this._formData;
+  }
+
+  public set formData(data: any) {
+    for (const key in this._formData) {
+      const newValue = data[key];
+      this._formDataSetter[key](newValue);
+      this.stateChanged.emit();
+    }
```

Comment:
Let's reduce the number of update by only triggering once the state update:

```suggestion
    for (const key in this._formData) {
      const newValue = data[key];
      this._formDataSetter[key](newValue);
    }
    this.stateChanged.emit();
```